web/client-api/src/main/java/io/deephaven/web/client/api/JsLayoutHints.java

```diff
@@ -51,6 +88,15 @@ public JsLayoutHints parse(String hints) {
             frozenColumns = JsObject.freeze(freezeStr.split(","));
         }
 
+        final String groupsStr = options.get("columnGroups");
+        if (groupsStr != null && !groupsStr.isEmpty()) {
+            List<ColumnGroup> groups =
+                    Arrays.stream(groupsStr.split("\\|")).map(ColumnGroup::new).map(JsObject::freeze)
+                            .collect(Collectors.toList());
+
+            columnGroups = JsObject.freeze(groups.toArray(new ColumnGroup[0]));
```

Comment:
skip the List and just call .toArray(ColumnGroup[]::new) on the stream itself, and freeze that